.github/workflows/sonar-checks.yml

```diff
@@ -11,14 +11,30 @@ jobs:
   sonarcloud:
     name: Sonar Cloud Static Scans
     runs-on: ubuntu-latest
+    env:
+        go_version: '1.23'
+        
     if: github.event.workflow_run.conclusion == 'success' && github.event.workflow_run.event == 'pull_request'
 
     steps:
-      - name: Fetch Code Coverage Report
-        uses: actions/download-artifact@v4
+      - name: Checkout Provider
+        uses: actions/checkout@v4
         with:
-          name: coverage-report
-          path: .
+          fetch-depth: 0
+
+      - name: Setup Go ${{ env.go_version }}
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.go_version }}
+
+      - name: Run unit tests with coverage
+        run: make testcov
```

Comment:
The `unit_test_and_quality.yml` workflow where we're moving this from has more steps before this (`go get`, `go build`. Do we need those steps here? If we don't, why do we have them in the other workflow?

Reply:
If something fails in those discrete steps during the initial PR workflow, it's a bit easier to see and know what's failing.  Once we get here, I don't see a point of splitting it all out again.  This is a stop gap measure to get this working without trying to submit more PRs to get the upload / download of the code coverage working.